assets/src/edit-story/components/canvas/useInsertElement.js

```diff
@@ -166,8 +168,8 @@ function getElementProperties(
   if (!isNum(y)) {
     y = 0;
   }
-  x = dataPixels(Math.min(x, PAGE_WIDTH - width));
-  y = dataPixels(Math.min(y, PAGE_HEIGHT - height));
+  x = dataPixels(ignorePageBoundary ? x : Math.min(x, PAGE_WIDTH - width));
+  y = dataPixels(ignorePageBoundary ? y : Math.min(y, PAGE_HEIGHT - height));
```

Comment:
Originally when I got the text set drag-n-drop working, I noticed if I placed a text set to close to a page border, the rendered text set's layout would get wonky (i.e. they'd overlap each other, alignment would change, etc...). So I added a prop that ignores the page boundaries so when a user drops a text set, it'll render as expected even if clips the edge of the safe zone/page.  I figure the user can then adjust the rendered text set's position as needed while keeping its layout intact.

Reply:
This can result in the user dropping the text sets out of the Page though -- probably it should still ensure fits on the page (e.g. adjust the `x` and `y` but ensuring the alignment is still all correct) or alternatively we should not insert the set at all if it's out of the boundaries.

We should avoid this happening:
![drag-drop](https://user-images.githubusercontent.com/3294597/93805873-9a03b300-fc48-11ea-8bda-6f9606430a29.gif)

Also, I wonder if using the native drag is suitable for the text sets -- it changes quite a lot after positioning. Using Moveable could also be an option to displays the dragged set the same way as it will be added on the Page as well. Might be good to confirm with Sam, WDYT?

Reply:
Okay, I fixed the drag and drop problem.  Now if you drag and drop a text set, it'll make sure to adjust itself to be within the boundaries of the page (then the user can adjust it to wherever they want).

And I talked to Sam, and she definitely wants the drag icon to have the same width/height as the rendered text set, so I'm working on that now.